src/TransactionInclusionProver.sol

```diff
@@ -23,44 +25,59 @@ contract TransactionInclusionProver is ITransactionInclusionProver {
 
     ITrustedOracle private _oracle;
 
+    /**
+     * @dev Initializes the contract with the address of the trusted oracle.
+     * @param oracleAddress The address of the trusted oracle.
+     */
     constructor(address oracleAddress) {
         _oracle = ITrustedOracle(oracleAddress);
     }
 
+    /**
+     * @dev Verifies that a transaction is included in a block.
+     * @param data The data needed to verify the transaction inclusion.
+     * @return A boolean indicating whether the transaction is included in the block.
+     */
     function proveTransactionInclusion(ProverDto memory data) external view returns (bool) {
         if (!data.txReceipt.status) return false;
 
         if (_oracle.getBlockHash(data.blockNumber) != _getBlockHash(data.blockData)) return false;
 
         bytes32 txReceiptHash = _getReceiptHash(data.txReceipt);
-        if (
-            MerkleProof.verify(
-                data.receiptProofBranch, Utils.bytesToBytes32(data.blockData.receiptsRoot, 0), txReceiptHash
-            )
-        ) {
+        if (MerkleProof.verify(data.receiptProofBranch, data.blockData.receiptsRoot, txReceiptHash)) {
```

Comment:
you can rewrite this to just `return MerkleProof.verify(data.receiptProofBranch, data.blockData.receiptsRoot, txReceiptHash)`